src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs

```diff
@@ -1614,6 +1615,27 @@ public virtual ObjectResult StatusCode(int statusCode, object value)
         public virtual UnauthorizedResult Unauthorized()
             => new UnauthorizedResult();
 
+        /// <summary>
+        /// Creates a <see cref="PartialViewResult"/> by specifying the name of a partial to render.
+        /// </summary>
+        /// <param name="partialViewName">The partial name.</param>
+        /// <returns>The created <see cref="PartialViewResult"/> object for the response.</returns>
+        public virtual PartialViewResult Partial(string partialViewName)
+            => Partial(partialViewName, model: null);
+
+        /// <summary>
+        /// Creates a <see cref="PartialViewResult"/> by specifying the name of a partial to render and the model object.
+        /// </summary>
+        /// <param name="partialViewName">The partial name.</param>
+        /// <param name="model">The model to be passed into the partial.</param>
+        /// <returns>The created <see cref="PartialViewResult"/> object for the response.</returns>
+        public virtual PartialViewResult Partial(string partialViewName, object model)
+            => new PartialViewResult
+            {
+                ViewName = partialViewName,
+                ViewData = new ViewDataDictionary(ViewData) { Model = model }
```

Comment:
If you follow the same pattern as controllers, you could get away without the need to new-ing this up: https://github.com/aspnet/Mvc/blob/master/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs#L148-L155

Reply:
I already did that before, but I'm the test fails in the CI, coz I'm unable to test it in my end because VS gives me an error when I run the tests .. anyhow I will try to follow the pattern